File: frontend/src/types/market.ts
export interface MarketQuote {
  ticker: string;
  current_price: number;
  previous_close: number;
  daily_change: number;
  daily_change_pct: number;
  week_52_high: number;
  week_52_low: number;
  market_cap: number | null;
  pe_ratio: number | null;
  dividend_yield: number;
  analyst_rating: string | null;
}

export interface PricePoint {
  date: string;
  open: number;
  close: number;
  high: number;
  low: number;
  volume: number;
}

File: backend/app/schemas/market.py
"""Pydantic schemas for market data responses."""

from datetime import date

from pydantic import BaseModel


class MarketQuote(BaseModel):
    """Schema for a real-time market quote."""

    ticker: str
    current_price: float
    previous_close: float
    daily_change: float
    daily_change_pct: float
    week_52_high: float
    week_52_low: float
    market_cap: float | None
    pe_ratio: float | None
    dividend_yield: float
    analyst_rating: str | None


class PricePoint(BaseModel):
    """Schema for a historical price data point."""

    date: date
    open: float
    close: float
    high: float
    low: float
    volume: float
